integrations/api_user_follow_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integrations
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	api "code.gitea.io/gitea/modules/structs"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAPIFollow(t *testing.T) {
+	defer prepareTestEnv(t)()
+
+	user1 := "user4"
+	user2 := "user1"
+
+	session1 := loginUser(t, user1)
+	token1 := getTokenForLoggedInUser(t, session1)
+
+	session2 := loginUser(t, user2)
+	token2 := getTokenForLoggedInUser(t, session2)
+
+	t.Run("Follow", func(t *testing.T) {
+		defer PrintCurrentTest(t)()
+
+		req := NewRequest(t, "PUT", fmt.Sprintf("/api/v1/user/following/%s?token=%s", user1, token2))
+		MakeRequest(t, req, http.StatusNoContent)
+	})
+
+	t.Run("ListFollowing", func(t *testing.T) {
+		defer PrintCurrentTest(t)()
+
+		req := NewRequest(t, "GET", fmt.Sprintf("/api/v1/users/%s/following?token=%s", user2, token2))
```

Comment:
should token1 be used here? As otherwise we could use `/api/v1/user/following?token=%s`

Reply:
That's intended so both endpoints are called.